api/ring-types.ts

```diff
@@ -179,7 +181,7 @@ export interface RingDeviceData {
   locked?: 'jammed' | 'locked' | 'unlocked' | 'unknown'
   roomId?: number
   volume?: number
-  mode?: AlarmMode
+  mode?: AlarmMode | ThermostatMode
```

Comment:
Outside the scope of this PR, but I briefly explored making `RingDeviceData` a [discriminated union](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions).

It’d allow us to mark fewer properties as optional (e.g. whenever `deviceType` is `temperature-control.thermostat`, then `setPoint` will exist). And it would allow us to clean-up `mode` (since no device actually supports all of `AlarmMode` and `ThermostatMode` values for `mode`).

I think we’d need to discriminate based on both `deviceType` and `categoryId`, in order to differentiate between e.g. `MultiLevelSwitch`, `MultiLevelBulb` and `Fan`.

Reply:
I agree that a discriminated union would be completely appropriate here.  The main difficulty in creating it will be getting accurate representations of the types for each individual accessory type.  Another problem is how to handle unknown device types since we don't know what their data will look like.  When I initially made this type, I started down that road but switched away just to save time and get something out the door.  Now could be a good time to re-address it in a separate PR.